src/opendr/perception/skeleton_based_action_recognition/continual_stgcn_learner.py

```diff
@@ -0,0 +1,698 @@
+        super(CoSTGCNLearner, self).__init__(
+            lr=lr,
+            iters=iters,
+            batch_size=batch_size,
+            optimizer=optimizer,
+            lr_schedule=lr_schedule,
+            backbone=backbone,
+            network_head=network_head,
+            temp_path=temp_path,
+            checkpoint_after_iter=checkpoint_after_iter,
+            checkpoint_load_iter=checkpoint_load_iter,
+            device=device,
+            threshold=0.0,
+        )
+
+        self.weight_decay = weight_decay
+        self.momentum = momentum
+        self.drop_last = drop_last
+        self.pin_memory = pin_memory
+        self.num_workers = num_workers
+        self.seed = seed
+        self.num_classes = num_classes
+        self.loss = loss
+        self._ort_session = None
+        self._ort_state = None
+        self.num_point = num_point
+        self.num_person = num_person
+        self.in_channels = in_channels
+        self.graph_type = graph_type
+        self.sequence_len = sequence_len
+
+        if self.graph_type is None:
+            raise ValueError(
+                self.graph_type + "is not a valid graph type. Supported graphs: ntu, openpose"
+            )
+        if self.backbone is None or self.backbone not in _MODEL_NAMES:
+            raise ValueError(
+                self.backbone + f"is not a valid dataset name. Supported methods: {_MODEL_NAMES}"
+            )
+
+        pl.seed_everything(self.seed)
+        self.init_model()
+
+    def init_model(self) -> Union[CoStGcnMod, CoAGcnMod, CoSTrMod]:
+        """Initialise model with random parameters
+
+        Returns:
+            Union[CoStGcnMod, CoAGcnMod, CoSTrMod]: model
+        """
+        Model = {
+            "costgcn": CoStGcnMod,
+            "coagcn": CoAGcnMod,
+            "costr": CoSTrMod,
+        }[self.backbone]
+
+        self.model = Model(
+            self.num_point,
+            self.num_person,
+            self.in_channels,
+            self.graph_type,
+            self.sequence_len,
+            self.num_classes,
+            self.loss,
+        ).to(device=self.device)
+        return self.model
+
+    def download(
+        self,
+        dataset_name="nturgbd_cv",
+        experiment_name="stgcn_nturgbd",
+        path=None,
+        method_name="costgcn",
+        mode="pretrained",
+        verbose=True,
+        url=OPENDR_SERVER_URL + "perception/skeleton_based_action_recognition/",
+        file_name="costgcn_ntu60_xview_joint.ckpt",
+    ):
+        if path is None:
+            path = self.parent_dir
+
+        if mode == "pretrained":
+            file_url = os.path.join(url, 'pretrained_models', method_name, file_name)
+            target_path = Path(path) / method_name / dataset_name / file_name
+            if target_path.exists():
+                return str(target_path)
+            target_path.parent.mkdir(exist_ok=True, parents=True)
+            print(f"Downloading {file_name} to {str(target_path)}")
```

Comment:
Print could go inside an if verbose block, since the verbose argument exists

Reply:
Added in 87a7416f